### ISSUE TITLE
[bug] false-positive with `use-memo`

### ISSUE THREAD

**Issue by nstepien** [Rel1cx/eslint-react#1734]
### Describe the bug

5.6.0 introduced false-positives in the `use-memo` rule

### Reproduction

https://github.com/Comcast/react-data-grid/pull/4036
We have this code in a `useMemo`, and the rule thinks the `width` variable is outside the `useMemo` callback.
```ts
    for (const column of columns) {
      let width = getColumnWidth(column);

      if (typeof width === 'number') {
        width = clampColumnWidth(width, column);
      } else {
        width = column.minWidth;
```

### Expected behavior

No false-positives

### Platform and versions

```Text
> npm version    
{
  'react-data-grid': '7.0.0-beta.59',
  npm: '11.13.0',
  node: '25.9.0',
  acorn: '8.16.0',
  ada: '3.4.4',
  amaro: '1.1.8',
  ares: '1.34.6',
  brotli: '1.2.0',
  cldr: '48.0',
  icu: '78.2',
  lief: '0.17.0',
  llhttp: '9.3.1',
  merve: '1.2.2',
  modules: '141',
  napi: '10',
  nbytes: '0.1.3',
  ncrypto: '0.0.1',
  nghttp2: '1.68.0',
  nghttp3: '',
  ngtcp2: '',
  openssl: '3.5.5',
  simdjson: '4.5.0',
  simdutf: '7.3.3',
  sqlite: '3.51.3',
  tz: '2026a',
  undici: '7.24.4',
  unicode: '17.0',
  uv: '1.51.0',
  uvwasi: '0.0.23',
  v8: '14.1.146.11-node.25',
  zlib: '1.3.1-e00f703',
  zstd: '1.5.7'
}

eslint@10.2.1
@eslint-react/eslint-plugin@5.6.0
```

### Stack trace

```Text

```
